components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/LogoutRequestBuilder.java

```diff
@@ -70,6 +82,47 @@ public LogoutRequest buildLogoutRequest(String subject,String sessionIndexId, St
     }
 
     /**
+     * Build the logout request
+     * @param subject name of the user
+     * @param reason reason for generating logout request.
+     * @return LogoutRequest object
+     */
+    public LogoutRequest buildSignedLogoutRequest(String subject,String sessionIndexId, String reason,
+            String issuerId, int tenantId, String tenantDomain, String destination) throws Exception {
```

Comment:
Fixed generic exception throwing in #1528 (https://github.com/wso2/carbon-apimgt/commit/c6d6cb237bd7654eda952fb2fbd74863e9899cfc)